relay-event-normalization/src/normalize/user_agent.rs

```diff
@@ -46,10 +46,48 @@ pub fn normalize_user_agent_info_generic(
     platform: &Annotated<String>,
     user_agent_info: &RawUserAgentInfo<&str>,
 ) {
-    if !contexts.contains::<BrowserContext>() {
+    // Check if a browser context already exists
+    let existing_browser_context = contexts.contains::<BrowserContext>();
+
+    if !existing_browser_context {
         if let Some(browser_context) = BrowserContext::from_hints_or_ua(user_agent_info) {
             contexts.add(browser_context);
         }
+    } else if let Some(browser_context) = contexts.get_mut::<BrowserContext>() {
+        // Browser context exists but might need backfilling from user_agent
+        if browser_context.user_agent.value().is_none() && user_agent_info.user_agent.is_some() {
+            browser_context.set_user_agent(user_agent_info.user_agent.unwrap());
+        }
+
+        // If the name is not set, but we have a user_agent, try to derive it
+        if browser_context.name.value().is_none() {
+            // First try to get it from the browser context's user_agent field if it exists
+            if let Some(browser_ua) = browser_context.user_agent.value() {
+                if let Some(parsed_context) = BrowserContext::parse_user_agent(browser_ua) {
+                    if browser_context.name.value().is_none() {
+                        browser_context.name = parsed_context.name;
+                    }
+                    if browser_context.version.value().is_none() {
+                        browser_context.version = parsed_context.version;
+                    }
+                }
+            }
+            // If still not set, try with the request interface's user_agent
+            else if let Some(request_ua) = user_agent_info.user_agent {
+                if let Some(parsed_context) = BrowserContext::parse_user_agent(request_ua) {
+                    if browser_context.name.value().is_none() {
+                        browser_context.name = parsed_context.name;
+                    }
+                    if browser_context.version.value().is_none() {
+                        browser_context.version = parsed_context.version;
+                    }
+                    // Also set user_agent for the browser context if it's missing
+                    if browser_context.user_agent.value().is_none() {
+                        browser_context.set_user_agent(request_ua);
+                    }
+                }
+            }
+        }
```

Comment:
I think in general the better approach is:

1. infer the user agent from the request and browser location (browser context should probably take precedence)
2. calculate a context based on the UA
3. merge the two contexts (existing + new), where the existing values have precedence. This should theoretically be possible with a `ProcessValue` visitor.

Reply:
`ProcessValue` won't work, there is a trick with `IntoValue` and `FromValue` but I think manually merging is the way to go.